components/802.15.4_RF/stm-s2lp-rf-driver/source/rf_configuration.c

```diff
@@ -181,5 +180,31 @@ uint32_t rf_conf_calculate_deviation(phy_modulation_index_e modulation_index, ui
     return deviation;
 }
 
+uint8_t rf_conf_calculate_pa_level_dbm(int8_t power_dbm)
+{
+#if MBED_CONF_S2LP_EXTERNAL_FRONTEND
+    if (power_dbm > 10) {
+        power_dbm = 10;
+    }
+#else
+    if (power_dbm > MAX_PA_VALUE) {
+        power_dbm = MAX_PA_VALUE;
+    }
+#endif
+
+    if (power_dbm < MIN_PA_VALUE) {
+        power_dbm = MIN_PA_VALUE;
+    }
+
+    uint8_t pa_level_value = 0;
+    if (power_dbm > 14) {
+        pa_level_value = 1;
```

Comment:
How did you figure out that writing 1 to PA_POWER registers equals to 14dBm? I couldn't find it from datasheet.

Reply:
Hi @JarkkoPaso I've got this piece of code from the official repository, as you can see here (line: 1173):
https://github.com/stm32duino/S2-LP/blob/e61561b8705a806466e0064eb148abb61fa74579/src/S2LP_Radio.cpp

On the datasheet my understanding is that when PA_POWER[PA_LEVEL_MAX_INDEX] is 0 PA is off.

Page 30. Item: 5.4.2 Amplitude modulation

> When OOK is selected, a bit '1' is transmitted with a programmed power, set by register PA_POWER[PA_LEVEL_MAX_INDEX], and a bit '0' is transmitted without output power (PA off) and specified by the register PA_POWER[0].

I know we are not using OOK modulation, but maybe it applies to FSK modulation as well. To be honest I've always used like this and always worked so I didn't looked twice. Maybe someone from ST can help us here. Maybe there's just something missing from datasheet, or just not as well explained there.